Makefile

```diff
@@ -1,19 +1,43 @@
 # holochain Makefile
 
+F=RUSTFLAGS="-Dwarnings"
+
 # mark everything as phony because it doesn't represent a file-system output
-.PHONY: default build-workspace test-workspace
+.PHONY: default \
+	static-all static-fmt static-toml static-clippy static-doc \
+	build-workspace test-workspace
 
 # default to running everything (first rule)
 default: build-workspace test-workspace
 
+# execute all static code validation
+static-all: static-fmt static-toml static-clippy static-doc
+
+# ensure committed code is formatted properly
+static-fmt:
+	$(F) cargo fmt --check
+
+# lint our toml files
+static-toml:
+	cargo install taplo-cli
+	$(F) taplo format --check ./*.toml
+	$(F) taplo format --check ./crates/**/*.toml
+
+# ensure our chosen style lints are followed
+static-clippy:
+	$(F) cargo clippy
```

Comment:
Can this be --all-targets and -D warnings?

Reply:
It's already -Dwarnings, that's what's in the `$(F)`. I tried adding --all-targets, but there are crazy tons of lints in our tests, so I added a todo to fix those and enable that flag at a later date.

Reply:
made a start: https://github.com/holochain/holochain/pull/3995 but ran out of daylight... @anyone feel free to pick this up before I get to it Monday.

Reply:
I'll pick this up, I forgot that the tests don't get linted currently. It's quite a few to fix! :)